lib/src/gotrue_client.dart

```diff
@@ -99,32 +100,35 @@ class GoTrueClient {
     String? phone,
     String? password,
     Provider? provider,
+    OpenIDConnectCredentials? oidc,
     AuthOptions? options,
   }) async {
     _removeSession();
 
-    if (email != null) {
-      if (password == null) {
-        final response = await api.sendMagicLinkEmail(email, options: options);
-        return GotrueSessionResponse(error: response.error);
-      } else {
-        return _handleEmailSignIn(email, password, options: options);
-      }
-    } else if (phone != null) {
-      if (password == null) {
-        final response = await api.sendMobileOTP(phone);
-        return GotrueSessionResponse(error: response.error);
-      } else {
-        return _handlePhoneSignIn(phone, password);
-      }
-    } else if (provider != null) {
+    if (email != null && password == null) {
+      final response = await api.sendMagicLinkEmail(email, options: options);
+      return GotrueSessionResponse(error: response.error);
+    }
+    if (email != null && password != null) {
+      return _handleEmailSignIn(email, password, options: options);
+    }
+    if (phone != null && password == null) {
+      final response = await api.sendMobileOTP(phone);
+      return GotrueSessionResponse(error: response.error);
+    }
+    if (phone != null && password != null) {
+      return _handlePhoneSignIn(phone, password);
+    }
```

Comment:
This should be how it was before

Reply:
I kind of disagree. The way it previously was, was full of nested if statements that were hard to follow. This is also how the gotrue-js does it.